### PR TITLE
[Blueprints] Update Export and Import cmdlets with improvements

### PR DESCRIPTION
**Pull request by filizt**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Blueprint/Blueprint/Az.Blueprint.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.2.0'
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/Az.Blueprint.psd1]: @filizt you can revert the changes made in this file -- we're version the module before release

File: src/Blueprint/Blueprint/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,9 @@
 -->
 ## Upcoming Release
 
+## Version 0.2.1
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/ChangeLog.md]: @filizt you can get rid of this header and put the below line below `## Upcoming Release`

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs [hunk 3]
```diff
@@ -149,8 +150,17 @@ protected string CreateFolderIfNotExist(string path, string folderName)
         {
             var folderPath = Path.Combine(path, folderName);
 
-            if (!AzureSession.Instance.DataStore.DirectoryExists(folderPath))
+            if (AzureSession.Instance.DataStore.DirectoryExists(folderPath))
             {
+                AzureSession.Instance.DataStore.EmptyDirectory(folderPath);
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs]: @filizt is there any reason we want to delete all of the files in this user's folder?

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs [hunk 4]
```diff
@@ -163,11 +173,9 @@ protected void ImportArtifacts(string blueprintName, string scope, string inputP
 
             var artifactsPath = GetValidatedFolderPath(inputPath, artifacts);
 
-            var artifactFiles = AzureSession.Instance.DataStore.GetFiles(artifactsPath, "*.json", SearchOption.TopDirectoryOnly);
+            if (artifactsPath == null) return; // if blueprint doesn't contain artifacts don't proceed.
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs]: @filizt nit:

```cs
if (artifactsPath == null)
{
    return;
}
```

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs [hunk 5]
```diff
@@ -48,38 +49,45 @@ public class ExportAzureRmBlueprint : BlueprintDefinitionCmdletBase
...
+                foreach (var artifact in artifacts)
+                {
+                    string serializedArtifact =
+                        BlueprintClient.GetBlueprintArtifactJsonFromObject(Blueprint.Scope, Blueprint.Name, artifact,
+                            Version);
 
-                var artifactFilePath = Path.Combine(artifactsPath, artifact.Name + ".json");
+                    var artifactFilePath = Path.Combine(artifactsPath, artifact.Name + ".json");
 
-                this.ConfirmAction(
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs]: @filizt do we not need this check `ConfirmAction` check anymore?

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs (hunk 3 above)
Comment (filizt) [reply src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs]: We wanted to make sure the the latest exported blueprint and its artifacts are the only blueprint and artifacts in the folder. So, if the user exported a blueprint and then made some changes(say deleted artifacts) through the portal and exported again, in the current implementation the export operation will keep the "deleted" artifact in the Artifacts folder since we are not doing any diffing between the local copy and the portal version of the blueprint. So, we thought we could delete the files in this case and re-create the blueprint and its artifact in this folder. That being said, if there is a way to avoid deleting files this way and still satisfy the above case, it'd be good to know. 

I guess we can always just delete the "Artifacts" folder, as a potentially less impactful option.

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs (hunk 5 above)
Comment (filizt) [reply src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs]: We don't since the "Artifacts" folder would have been deleted in the first blueprint creation step, which is the first confirm action, should it already existed locally.

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs (hunk 3 above)
Comment (filizt) [reply src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs]: We also prompt the user of the folder deletion in the caller of the CreateFolderIfNotExist function here: https://github.com/Azure/azure-powershell/blob/7c8d702511580458b22c28506b4e7f18110155fa/src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs#L62
